### PR TITLE
Changed query param support from String values to Object

### LINKED ISSUE #162: Support queryParam(String, Object)
and call toString on the value

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #162

### CODE REVIEW

File: riptide-core/src/test/java/org/zalando/riptide/RequesterTest.java [hunk 1]
```diff
@@ -88,14 +88,15 @@ public void shouldEncodeInlinedQueryParams() {
 
     @Test
     public void shouldAppendQueryParams() {
-        server.expect(requestTo("https://api.example.com?foo=bar&foo=baz&bar=null"))
+        server.expect(requestTo("https://api.example.com?foo=true&foo=null&foo=bar&bar=123"))
```
Comment (jhorstmann) [inline riptide-core/src/test/java/org/zalando/riptide/RequesterTest.java]: Serializing `null` as string `"null"` might be unexpected, but seems this was already done the same way before this change.

Comment (whiskeysierra) [inline riptide-core/src/test/java/org/zalando/riptide/RequesterTest.java]: Indeed, that was the same before. What would you expect? An empty string?

Comment (whiskeysierra) [inline riptide-core/src/test/java/org/zalando/riptide/RequesterTest.java]: With quotes? oO

Comment (lukasniemeier-zalando) [inline riptide-core/src/test/java/org/zalando/riptide/RequesterTest.java]: > With quotes? oO

No without :see_no_evil: but whatever we do here is probably unexpected from someone.

**Comment by lukasniemeier-zalando** [comment]
We've discussed this issue again and we will stick with `String` for now to prevent any API confusion.

**Pull request closed without merging**